src/components/axes/abstractAxis.ts

```diff
@@ -488,16 +488,43 @@ export module Axis {
                                     });
       var lastLabelClientRect: ClientRect;
 
-      visibleTickLabels.each(function (d: any) {
+      // Base interval value between labels
+      var interval = 1;
+
+      // Get the widths of all the labels
+      var visibleTickLabelRects: ClientRect[] = [];
+      visibleTickLabels.each(function(d: any) {
         var clientRect = this.getBoundingClientRect();
+        visibleTickLabelRects.push(clientRect);
+      });
+
+      // Figure out the appropriate interval for overlap
+      var hasSkipped = false;
+      for (var i = 1; i < visibleTickLabelRects.length; i++) {
+        // Checks to see if there is any overlap between the first element and the current element
+        if (!_Util.DOM.boxesOverlap(visibleTickLabelRects[i], visibleTickLabelRects[0])) {
+          // We increment the variable if the next label's width is greater than the previous one's (one-time)
+          // This only applies to the bottom and top orientations
+          if (visibleTickLabelRects[i].width > visibleTickLabelRects[i-1].width && !hasSkipped &&
+            (this._orientation === "bottom" || this._orientation === "top")) {
+            interval += 1;
+            hasSkipped = true;
```

Comment:
Sorry, clarification. They're not technically overlapping but are close enough to make visibility an issue (0-2px distance between labels).